test/lit/wat-kitchen-sink.wast

```diff
@@ -701,6 +701,112 @@
   drop
  )
 
+ ;; CHECK:      (func $block (type $void)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (block $l
+ ;; CHECK-NEXT:   (nop)
+ ;; CHECK-NEXT:   (nop)
+ ;; CHECK-NEXT:   (nop)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $block
+  nop
+  block $l (type $void)
+   nop
+   nop
+   nop
+  end $l
+ )
+
+ ;; CHECK:      (func $block-folded (type $void)
+ ;; CHECK-NEXT:  (local $scratch (i32 i32))
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (local.set $scratch
+ ;; CHECK-NEXT:   (block $l (result i32 i32)
+ ;; CHECK-NEXT:    (nop)
+ ;; CHECK-NEXT:    (nop)
+ ;; CHECK-NEXT:    (unreachable)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (tuple.extract 0
+ ;; CHECK-NEXT:    (local.get $scratch)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (tuple.extract 1
+ ;; CHECK-NEXT:    (local.get $scratch)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (unreachable)
+ ;; CHECK-NEXT: )
+ (func $block-folded
+  nop
+  (block $l (result i32) (result) (result i32)
```

Comment:
this `(result)` is... valid? :scream:  I guess it has to be unless the spec special cases it...

Reply:
Yes, and even worse, you can put `(result)` literally anywhere in a wat file because it is an abbreviation that is entirely rewritten away. We only support it in reasonable places, though.